gym/vector/async_vector_env.py

```diff
@@ -0,0 +1,482 @@
+                        and (self._num_retries >= self.max_retries):
+                    for pipe in self.parent_pipes:
+                        pipe.recv()
+                    logger.error('The maximum number of retries has been '
+                        'reached. Raising the last exception back to the main '
+                        'process.')
+                    raise exctype(value)
+
+                logger.warn('Restarting Worker-{0}...'.format(index))
+                process, parent_pipe = self._start_process(index)
+                parent_pipe.send(('_restart', reset))
+
+                self.processes[index] = process
+                self.parent_pipes[index] = parent_pipe
+                self._num_retries += 1
+
+    def _start_process(self, index):
+        target = _worker_shared_memory if self.shared_memory else _worker
+        parent_pipe, child_pipe = self.ctx.Pipe()
+        process = self.ctx.Process(target=target,
+            name='Worker<{0}>-{1}'.format(type(self).__name__, index),
+            args=(index, CloudpickleWrapper(self.env_fns[index]), child_pipe,
+            parent_pipe, self._obs_buffer, self.error_queue, self.episodic))
+
+        process.deamon = True
+        process.start()
+        child_pipe.close()
+
+        return process, parent_pipe
+
+    def __del__(self):
+        if hasattr(self, 'closed'):
+            if not self.closed:
+                self.close(terminate=True)
+
+
+def _worker(index, env_fn, pipe, parent_pipe, shared_memory, error_queue,
+            episodic):
+    assert shared_memory is None
+    env = env_fn()
+    _zero_observation = create_empty_array(env.observation_space, n=None,
+                                           fn=np.zeros)
+    episode_done = False
+    parent_pipe.close()
+    try:
+        while True:
+            command, data = pipe.recv()
+            if command == 'reset':
+                observation = env.reset()
+                episode_done = False
+                pipe.send(observation)
+            elif command == 'step':
+                if episodic and episode_done:
+                    observation = _zero_observation
+                    reward, done, info = 0., True, {}
+                else:
+                    observation, reward, done, info = env.step(data)
+                    if done:
+                        if episodic:
+                            observation = _zero_observation
+                            episode_done = True
+                            info.update({'AsyncVectorEnv.end_episode': True})
+                        else:
+                            observation = env.reset()
+                pipe.send((observation, reward, done, info))
+            elif command == 'seed':
+                env.seed(data)
+                pipe.send(None)
+            elif command == 'close':
+                pipe.send(None)
+                break
+            elif command == '_check_observation_space':
+                try:
```

Comment:
This seems odd, can this check be improved?

Reply:
This was addressed in #1519, I will fix this now that this PR has been merged